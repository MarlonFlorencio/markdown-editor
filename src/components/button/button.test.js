'use strict'

import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Button from './index'

const noop = () => null

it('Should Button default to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={noop}>
      Click me
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button success to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={noop} kind='success'>
      Click me Success
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Should Button success to match snapshot', () => {
  const tree = renderer.create(
    <Button onClick={noop} kind='danger'>
      Click me Danger
    </Button>
  ).toJSON()
  expect(tree).toMatchSnapshot()
})

it('Button with prop kind="success" should has class "-success"', () => {
  const wrapper = shallow(
    <Button onClick={noop} kind='success'>
      Button Success
    </Button>
  )
  expect(wrapper.hasClass('-success')).toBe(true)
  expect(wrapper.hasClass('-danger')).toBe(false)
})

it('Button with prop kind="danger" should has class "-danger"', () => {
  const wrapper = shallow(
    <Button onClick={noop} kind='danger'>
      Button Success
    </Button>
  )
  expect(wrapper.hasClass('-danger')).toBe(true)
  expect(wrapper.hasClass('-success')).toBe(false)
})
