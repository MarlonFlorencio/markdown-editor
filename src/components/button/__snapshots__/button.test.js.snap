// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should Button default to match snapshot 1`] = `
<button
  className="button"
  onClick={[Function]}
>
  Click me
</button>
`;

exports[`Should Button success to match snapshot 1`] = `
<button
  className="button -success"
  onClick={[Function]}
>
  Click me Success
</button>
`;

exports[`Should Button success to match snapshot 2`] = `
<button
  className="button -danger"
  onClick={[Function]}
>
  Click me Danger
</button>
`;
